neo/Consensus/ConsensusService.cs

```diff
@@ -277,262 +364,39 @@ private void OnPersistCompleted(Block block)
-                                              Blockchain.TimePerBlock.TotalSeconds * Math.Pow(2, message.ViewNumber));
-                }
-
-                InitializeConsensus(message.ViewNumber);
-            }
-
-            for (int i = 0; i < context.Validators.Length; i++)
-                if (verifiedChangeViewPayloads[i] != null)
-                    context.ChangeViewPayloads[i] = verifiedChangeViewPayloads[i];
-
-            if (prepareRequestPayload != null)
-            {
-                Log($"regenerating prepare request");
-                // Note: If our node is the primary this will accept its own previously sent prepare request here.
-                OnPrepareRequestReceived(prepareRequestPayload, prepareRequestMessage);
+                if (context.State.HasFlag(ConsensusState.CommitSent))
+                    return;
+                ConsensusPayload[] changeViewPayloads = GetChangeViewPayloadsFromRecoveryMessage(context, payload, message);
+                foreach (ConsensusPayload changeViewPayload in changeViewPayloads)
+                    ReverifyAndProcessPayload(changeViewPayload);
             }
-
-            if (prepareResponses.Count > 0)
+            if (message.ViewNumber != context.ViewNumber) return;
+            if (!context.State.HasFlag(ConsensusState.CommitSent))
             {
-                Log($"regenerating preparations: {prepareResponses.Count}");
-                foreach (var (prepareRespPayload, prepareResp) in prepareResponses)
-                    if (prepareRespPayload.ValidatorIndex != context.MyIndex)
-                        OnPrepareResponseReceived(prepareRespPayload, prepareResp);
+                ConsensusPayload prepareRequestPayload = GetPrepareRequestPayloadFromRecoveryMessage(context, payload, message);
+                if (prepareRequestPayload != null && !context.State.HasFlag(ConsensusState.RequestSent) && !context.State.HasFlag(ConsensusState.RequestReceived))
+                    ReverifyAndProcessPayload(prepareRequestPayload);
+                ConsensusPayload[] prepareResponsePayloads = GetPrepareResponsePayloadsFromRecoveryMessage(context, payload, message, prepareRequestPayload);
+                foreach (ConsensusPayload prepareResponsePayload in prepareResponsePayloads)
+                    ReverifyAndProcessPayload(prepareResponsePayload);
             }
-
-            RestoreCommits(message);
+            ConsensusPayload[] commitPayloads = GetCommitPayloadsFromRecoveryMessage(context, payload, message);
+            foreach (ConsensusPayload commitPayload in commitPayloads)
+                ReverifyAndProcessPayload(commitPayload);
         }
 
         private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest message)
         {
-            if (context.State.HasFlag(ConsensusState.RequestReceived)) return;
+            if (context.State.HasFlag(ConsensusState.RequestSent) || context.State.HasFlag(ConsensusState.RequestReceived))
```

Comment:
I see, this allows the Primary to replay it's previously generated `PrepareRequest` instead of needing to set both the flags. I think this change could be made on the existing code and avoid that without needing to make